stan/math/opencl/prim/inv_logit.hpp

```diff
@@ -0,0 +1,63 @@
+#ifndef STAN_MATH_OPENCL_PRIM_FUN_INV_SQUARE_HPP
+#define STAN_MATH_OPENCL_PRIM_FUN_INV_SQUARE_HPP
+#ifdef STAN_OPENCL
+
+#include <stan/math/opencl/matrix_cl.hpp>
+#include <stan/math/opencl/kernel_generator.hpp>
+#include <stan/math/prim/fun/constants.hpp>
+
+namespace stan {
+namespace math {
+/**
+ * Returns the inverse logit function applied to the kernel generator
+ expression.
+ *
+ * The inverse logit function is defined by
+ *
+ * \f$\mbox{logit}^{-1}(x) = \frac{1}{1 + \exp(-x)}\f$.
+ *
+ * This function can be used to implement the inverse link function
+ * for logistic regression.
+ *
+ * The inverse to this function is <code>logit</code>.
+ *
+ \f[
+ \mbox{inv\_logit}(y) =
+ \begin{cases}
+ \mbox{logit}^{-1}(y) & \mbox{if } -\infty\leq y \leq \infty \\[6pt]
+ \textrm{NaN} & \mbox{if } y = \textrm{NaN}
+ \end{cases}
+ \f]
+
+ \f[
+ \frac{\partial\, \mbox{inv\_logit}(y)}{\partial y} =
+ \begin{cases}
+ \frac{\partial\, \mbox{logit}^{-1}(y)}{\partial y} & \mbox{if } -\infty\leq
+ y\leq \infty \\[6pt] \textrm{NaN} & \mbox{if } y = \textrm{NaN} \end{cases} \f]
+
+ \f[
+ \mbox{logit}^{-1}(y) = \frac{1}{1 + \exp(-y)}
+ \f]
+
+ \f[
+ \frac{\partial \, \mbox{logit}^{-1}(y)}{\partial y} =
+ \frac{\exp(y)}{(\exp(y)+1)^2} \f]
+ *
+ * @tparam T_x type of input kernel generator expression x
+ * @param x input kernel generator expression.
+ * @return inverse logit of each value in x.
+ */
+template <typename T_x,
+          typename
+          = require_all_valid_kernel_expressions_and_none_scalar_t<T_x>>
+inline auto inv_logit(T_x&& x) {  // NOLINT
+  return select(
+      (x < 0.0),
+      select((x < LOG_EPSILON), exp(x), elewise_division(exp(x), 1.0 + exp(x))),
+      elewise_division(1.0, 1.0 + exp(-x)));
```

Comment:
x is used in multiple expressions here. If a rvalue is passed to this function x will likely go out of scope before the expression is evaluated. In other words we have the same issue as #1775, except with kernel generator expressions. 

You can either remove this function for now or evaluate the expression in the function.

Reply:
Thanks. I will do it with the device function then.